packages/stock-location/src/services/stock-location.ts

```diff
@@ -6,18 +6,12 @@ import {
   IEventBusService,
```

Comment:
thought(non-blocking, this-file): I am on board with the idea of introducing a context param to pass around managers. Though, I think the `execute` method and how it's called is very repetitive and therefore prone to errors. Ideally, this would be abstracted away - similar to the atomic phase in the core.

You might have already discussed how to solve this? 

I am fine keeping it for a follow up PR, but I want to make sure we look at that in the near future.

Reply:
Yes, as I suggested, the transaction pr can introduce a decorator to take care of the flow and that way there will always be the right manager available or the method will be wrap in a transaction. I ve already show this approach with the actual pattern and it can be reproduce with the new pattern